### ISSUE TITLE
z3: not installing z3_algebraic.h and other headers

### ISSUE THREAD

**Issue by jwiegley** [NixOS/nixpkgs#6139]
These headers are in the tarball being downloaded, they are just not getting copied into the right place.  Austin, I'll take a look at this in a few days if you don't already know the fix.  It mainly prevents the `haskell-z3` library from building.

Pinging @thoughtpolice

**Comment by michalpalka** [comment]
I managed to install libz3 and compile the Haskell z3 package with it. My changes are in this commit: michalpalka/nixpkgs@24ab87886c3c6098c06c8d570c8dac75a4871f4f . I needed to use fixDarwinDylibNames to be able to link against libz3 on OS X (I only tested this on OS X by the way). Note that this patch is based on me making educated guesses, and should be reviewed by somebody competent in nix/nixpkgs.
